src/resources/views/inc/form_save_buttons.blade.php

```diff
@@ -23,4 +23,88 @@
     </div>
 
     <a href="{{ url($crud->route) }}" class="btn btn-default"><span class="fa fa-ban"></span> &nbsp;{{ trans('backpack::crud.cancel') }}</a>
-</div>
\ No newline at end of file
+</div>
+
+@push('crud_fields_scripts')
+<script>
+    jQuery('document').ready(function($){
+
+      // Save button has multiple actions: save and exit, save and edit, save and new
+      var saveActions = $('#saveActions'),
+      crudForm        = saveActions.parents('form'),
+      saveActionField = $('[name="save_action"]');
+
+      saveActions.on('click', '.dropdown-menu a', function(){
+          var saveAction = $(this).data('value');
+          saveActionField.val( saveAction );
+          crudForm.submit();
+      });
+
+      // Ctrl+S and Cmd+S trigger Save button click
+      $(document).keydown(function(e) {
+          if ((e.which == '115' || e.which == '83' ) && (e.ctrlKey || e.metaKey))
+          {
+              e.preventDefault();
+              // alert("Ctrl-s pressed");
+              $("button[type=submit]").trigger('click');
+              return false;
+          }
+          return true;
+      });
+
+      // Place the focus on the first element in the form
+      @if( $crud->autoFocusOnFirstField )
+        @php
+          $focusField = array_first($fields, function($field) {
+              return isset($field['auto_focus']) && $field['auto_focus'] == true;
+          });
+        @endphp
+
+        @if ($focusField)
+          window.focusField = $('[name="{{ $focusField['name'] }}"]').eq(0),
+        @else
+          var focusField = $('form').find('input, textarea, select').not('[type="hidden"]').eq(0),
+        @endif
+
+        fieldOffset = focusField.offset().top,
+        scrollTolerance = $(window).height() / 2;
+
+        focusField.trigger('focus');
+
+        if( fieldOffset > scrollTolerance ){
+            $('html, body').animate({scrollTop: (fieldOffset - 30)});
+        }
+      @endif
+
+      // Add inline errors to the DOM
+      @if ($crud->inlineErrorsEnabled() && $errors->any())
+
+        window.errors = {!! json_encode($errors->messages()) !!};
+        // console.error(window.errors);
+
+        $.each(errors, function(property, messages){
+
+            var field = $('[name="' + property + '"]'),
+                container = field.parents('.form-group');
+
+            container.addClass('has-error');
+
+            $.each(messages, function(key, msg){
+                // highlight the input that errored
+                var row = $('<div class="help-block">' + msg + '</div>');
+                row.appendTo(container);
+
+                // highlight its parent tab
+                @if ($crud->tabsEnabled())
+                var tab_id = $(container).parent().attr('id');
```

Comment:
Indentation on this line and next seems wrong. (i.e. should be indented because of the @if @endif).